Sources/Jetworking/Client.swift

```diff
@@ -0,0 +1,152 @@
+        let request: URLRequest = createRequest(forHttpMethod: .POST, andPathComponent: endpoint.pathComponent)
+        let bodyData: Data = try! clientConfiguration.encoder.encode(body) // TODO: remove force unwrap --> Error handling
+        let dataTask = session.uploadTask(with: request, from: bodyData) { [weak self] data, urlResponse, error in
+            self?.handleResponse(data: data, urlResponse: urlResponse, error: error, endpoint: endpoint, completion: completion)
+        }
+
+        dataTask.resume()
+    }
+
+    public func put<BodyType: Codable, ResponseType>(endpoint: Endpoint<ResponseType>, body: BodyType, _ completion: @escaping (Result<ResponseType, Error>) -> Void) {
+        let request: URLRequest = createRequest(forHttpMethod: .PUT, andPathComponent: endpoint.pathComponent)
+        let bodyData: Data = try! clientConfiguration.encoder.encode(body)
+        let dataTask = session.uploadTask(with: request, from: bodyData) { [weak self] data, urlResponse, error in
+            self?.handleResponse(data: data, urlResponse: urlResponse, error: error, endpoint: endpoint, completion: completion)
+        }
+
+        dataTask.resume()
+    }
+
+    public func patch<BodyType: Codable, ResponseType>(endpoint: Endpoint<ResponseType>, body: BodyType, _ completion: @escaping (Result<ResponseType, Error>) -> Void) {
+        let request: URLRequest = createRequest(forHttpMethod: .PATCH, andPathComponent: endpoint.pathComponent)
+        let bodyData: Data = try! clientConfiguration.encoder.encode(body)
+        let dataTask = session.uploadTask(with: request, from: bodyData) { [weak self] data, urlResponse, error in
+            self?.handleResponse(data: data, urlResponse: urlResponse, error: error, endpoint: endpoint, completion: completion)
+        }
+
+        dataTask.resume()
+    }
+
+    public func delete<ResponseType>(endpoint: Endpoint<ResponseType>, parameter: [String: Any] = [:], _ completion: @escaping (Result<ResponseType, Error>) -> Void) {
+        let request: URLRequest = createRequest(forHttpMethod: .DELETE, andPathComponent: endpoint.pathComponent)
+        let dataTask = session.dataTask(with: request) { [weak self] data, urlResponse, error in
+            self?.handleResponse(data: data, urlResponse: urlResponse, error: error, endpoint: endpoint, completion: completion)
+        }
+
+        dataTask.resume()
+    }
+
+    private func createRequest(forHttpMethod httpMethod: HTTPMethod, andPathComponent pathComponent: String) -> URLRequest {
+        return .init(
+            url: clientConfiguration.baseURL.appendingPathComponent(pathComponent),
+            httpMethod: httpMethod,
+            headerFields: getHeaderFields()
+        )
+    }
+
+    private func getHeaderFields() -> [String: String] {
+        var headerFields: [String: String] = [
+            "Accept": "application/json",
+            "Content-Type": "application/json"
```

Comment:
We should make the default header fields exchangeable.
What do you think abound placing them in the ClientConfiguration together with a static default value which is returning this map? Then we could adjust them in the initializer if we need to?

Reply:
In my opinion we should tackle this in a future ticket. This MR just adds a first version of the framework. I will create tickets later on where we can discuss the different tasks. Maybe we only need some header fields in some cases and not always etc.